package/MDAnalysis/analysis/base.py

```diff
@@ -332,65 +401,300 @@ def _setup_frames(self, trajectory, start=None, stop=None, step=None,
+            Turn on verbosity
+
+        progressbar_kwargs : dict, optional
+            ProgressBar keywords with custom parameters regarding progress bar
+            position, etc; see :class:`MDAnalysis.lib.log.ProgressBar`
+            for full list.
+
+
+        .. versionadded:: 2.8.0
+        """
+        logger.info("Choosing frames to analyze")
+        # if verbose unchanged, use class default
+        verbose = getattr(self, "_verbose", False) if verbose is None else verbose
+
+        frames = indexed_frames[:, 1]
+
+        logger.info("Starting preparation")
+        self._prepare_sliced_trajectory(slicer=frames)
+        self._prepare()
+        if len(frames) == 0:  # if `frames` were empty in `run` or `stop=0`
+            return self
+
+        for idx, ts in enumerate(ProgressBar(
+                self._sliced_trajectory,
+                verbose=verbose,
+                **progressbar_kwargs)):
+            self._frame_index = idx  # accessed later by subclasses
+            self._ts = ts
+            self.frames[idx] = ts.frame
+            self.times[idx] = ts.time
+            self._single_frame()
+        logger.info("Finishing up")
+        return self
+
+    def _setup_computation_groups(
+        self, n_parts: int,
+        start: int = None, stop: int = None, step: int = None,
+        frames: Union[slice, np.ndarray] = None
+    ) -> list[np.ndarray]:
+        """
+        Splits the trajectory frames, defined by ``start/stop/step`` or
+        ``frames``, into ``n_parts`` even groups, preserving their indices.
+
+        Parameters
+        ----------
+        n_parts : int
+            number of parts to split the workload into
+        start : int, optional
+            start frame
+        stop : int, optional
+            stop frame
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; ``frames`` can
+            only be used *instead* of ``start``, ``stop``, and ``step``. Setting
+            *both* ``frames`` and at least one of ``start``, ``stop``, ``step``
+            to a non-default value will raise a :exc:`ValueError`.
+
+        Raises
+        ------
+        ValueError
+            if *both* ``frames`` and at least one of ``start``, ``stop``, or
+            ``frames`` is provided (i.e., set to another value than ``None``)
+
+        Returns
+        -------
+        computation_groups : list[np.ndarray]
+            list of (n, 2) shaped np.ndarrays with frame indices and numbers
+
+
+        .. versionadded:: 2.8.0
+        """
+        if frames is None:
+            start, stop, step = self._trajectory.check_slice_indices(start, stop, step)
+            used_frames = np.arange(start, stop, step)
+        elif not all(opt is None for opt in [start, stop, step]):
+            raise ValueError("start/stop/step cannot be combined with frames")
```

Comment:
Any reason why this isn't covered by tests? If not, then at the very least please raise an issue for it to be done later.

Reply:
Raised an issue: https://github.com/MDAnalysis/mdanalysis/issues/4648